### ISSUE TITLE
Regression on List tool

### ISSUE THREAD

**Issue by mems** [gskinner/regexr#137]
0e16ccc cause a regression on List tool.

## How to reproduce

Regex:
```javascript
/(https?[^'"]*)/g
```

Text:
```
{
src: 'https://example1.com/img.png'
},
{
src: 'https://example2.com/img.png'
},
{
src: 'https://example3.com/img.png'
}
```
Pattern (Tool/List):
```
$1\n
```
Result:
```
https://example1.com/img.png
https://example2.com/img.png
```
## Expected result
```
https://example1.com/img.png
https://example2.com/img.png
https://example3.com/img.png
```

## Simplified test case

```html
<script>
// inputs from regexr:
var regex = /(https?[^'"]*)//*global is not used here*/;
var source = `{
src: 'https://example1.com/img.png'
},
{
src: 'https://example2.com/img.png'
},
{
src: 'https://example3.com/img.png'
}`;
var str = "$1\n";
//
var repl, ref, result = "", lastIndex = -1, trimR = 0;
// https://github.com/gskinner/regexr/blob/0e16ccc06dc5016a0c0372591a5ae300b97a703b/js/views/DocView.js#L616-L625
//--
while (true) {
	ref = source.replace(regex, "\b");
	var index = ref.indexOf("\b");
	if (index === -1 || index === lastIndex) { break; }
	repl = source.replace(regex, str);
	result += repl.substr(index, repl.length-ref.length+1);
	source = ref.substr(index+1);
	lastIndex = index;
}
if (trimR) { result = result.substr(0,result.length-trimR); }
//--
console.log(result);
</script>
```

**Comment by gskinner** [comment]
Good catch. Thanks for the robust issue write up.

Fixed via 0e16ccc. The fix should be live shortly.

**Comment by wdamien** [comment]
@mems  thanks for the report! This fix is now live on regexr.com
